patches/v2024.5.0.patch

```diff
@@ -832,3 +832,15 @@ index 00ab2ff717..0950b9d787 100644
  
    --color-secondary-100: 240 240 240;
    --color-secondary-300: 206 212 220;
+diff --git a/apps/web/tsconfig.json b/apps/web/tsconfig.json
+index 543d7f25..5579906e 100644
+--- a/apps/web/tsconfig.json
++++ b/apps/web/tsconfig.json
+@@ -30,7 +30,6 @@
+   "files": [
+     "src/polyfills.ts",
+     "src/main.ts",
+-    "../../bitwarden_license/bit-web/src/main.ts",
```

Comment:
This isn't needed anymore as this is already changed upstream https://github.com/bitwarden/clients/pull/8818/files

So, I also do not see the huge benefit here either. 
We already build the OSS version, which is determined by Bitwarden to be GPLv3 compatibel ans not using there Bitwarden Licensed code.

Reply:
I think the benefit would be the assurance that the compiled code is actually `GPLv3` compatible and not relying on the content in the `bitwarden_license` directory at all to be built, which might not always be the case. (E.g. in [`web-v2023.8.0`](https://github.com/bitwarden/clients/blob/web-v2023.8.0/apps/web/src/app/billing/organizations/secrets-manager/sm-subscribe.component.ts#L10) to [`web-v2023.9.2`](https://github.com/bitwarden/clients/blob/web-v2023.9.2/apps/web/src/app/billing/organizations/secrets-manager/sm-subscribe.component.ts#L10) the secrets manager logo was technically included as part of the compilation when it was located in the `bitwarden_license` directory) 

It also enables third party vendors to be able to ship the source code without bundling the non-free code into their packages as explained in #162. (If we were to create a soft fork, we could also easily remove the non-free code from our repository to be on the safe side.)

Reply:
Precisely.  For distros and probably others all assurances should be taken that no non-OSI code ends up in the distro by mistake.  That said, it's standard practice to "rm -rf" anything you want to make sure isn't included or built against in RPM spec files.

My intent is to use the published/built sources for web vault since it's not binary, which means I can't rm -rf at build time with a Fedora-specific patch.

Since the patch does no harm and removes what's not being used anyway, it addresses the concern by having multiple things that would have to fail for the code to get into the build by mistake, while having 0 impact on existing functionality/expectations.

Reply:
I'm for the soft fork option but not currently in this repo. Though i still wonder if it isn't more work with upstream rebases etc. (But that is something i haven't looked at actually)

In general I agree that if possible without too much hassle to try and keep clear of any possible GPLv3 invalid build.

But seeing that upstream already made some changes regarding this i rather wait for a new release with those changes incorporated.